firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -106,25 +108,28 @@ public static FirebaseInstallations getInstance(@NonNull FirebaseApp app) {
     return (FirebaseInstallations) app.get(FirebaseInstallationsApi.class);
   }
 
+  /** Returns the application id of the {@link FirebaseApp} of this {@link FirebaseInstallations} */
+  @VisibleForTesting
+  String getApplicationId() {
+    return firebaseApp.getOptions().getApplicationId();
+  }
+
+  /** Returns the nick name of the {@link FirebaseApp} of this {@link FirebaseInstallations} */
+  @VisibleForTesting
+  String getName() {
+    return firebaseApp.getName();
+  }
+
   /**
    * Returns a globally unique identifier of this Firebase app installation. This is a url-safe
    * base64 string of a 128-bit integer.
    */
   @NonNull
   @Override
   public Task<String> getId() {
-    return getId(null);
-  }
-
-  /**
-   * Returns a globally unique identifier of this Firebase app installation.Also, updates the {@link
-   * AwaitListener} when the FID registration is complete.
-   */
-  private Task<String> getId(AwaitListener awaitListener) {
-    return Tasks.call(executor, this::getPersistedFid)
-        .continueWith(orElse(this::createAndPersistNewFid))
-        .onSuccessTask(
-            persistedFidEntry -> registerFidIfNecessary(persistedFidEntry, awaitListener));
+    Task<String> task = addGetIdListener();
+    executor.execute(doRegistration());
```

Comment:
we can make the runnable that doRegistration returns a field that just gets resubmitted; no need to instantiate a new runnable every time if all the state is contained in the parent

Reply:
Done. PTAL